cli/replicasets/list.go

```diff
@@ -0,0 +1,138 @@
+package replicasets
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/context"
+)
+
+const (
+	instanceMarker       = "•"
+	leaderInstanceMarker = "★"
+)
+
+func List(ctx *context.Ctx, args []string) error {
+	var err error
+
+	if err := FillCtx(ctx); err != nil {
+		return err
+	}
+
+	conn, err := connectToSomeJoinedInstance(ctx)
+	if err != nil {
+		return err
+	}
+
+	topologyReplicasets, err := getTopologyReplicasets(conn)
+	if err != nil {
+		return fmt.Errorf("Failed to get current topology replica sets: %s", err)
+	}
+
+	replicasetsSummary := getTopologyReplicasetsSummary(topologyReplicasets)
+
+	log.Infof("Current replica sets:\n%s", replicasetsSummary)
+
+	return nil
+}
+
+func getTopologyReplicasetsSummary(topologyReplicasets *TopologyReplicasets) string {
+	replicasetsSummaryByAliases := make(map[string]string)
+	replicasetsAliases := make([]string, len(*topologyReplicasets))
+
+	// sort replicasets summaries by aliases
+
+	i := 0
+	for _, topologyReplicaset := range *topologyReplicasets {
+		replicasetSummary := getTopologyReplicasetSummary(topologyReplicaset)
+		replicasetsSummaryByAliases[topologyReplicaset.Alias] = replicasetSummary
+		replicasetsAliases[i] = topologyReplicaset.Alias
+		i++
+	}
+
+	sort.Strings(replicasetsAliases)
+
+	replicasetsSummary := make([]string, len(replicasetsSummaryByAliases))
+	for i, replicasetAlias := range replicasetsAliases {
+		replicasetsSummary[i] = replicasetsSummaryByAliases[replicasetAlias]
+	}
+
+	return strings.Join(replicasetsSummary, "\n")
+}
+
+func getTopologyReplicasetSummary(topologyReplicaset *TopologyReplicaset) string {
+	replicasetSummary := []string{}
+
+	replicasetTitle := fmt.Sprintf(
+		"• %s",
+		common.ColorHiMagenta.Sprintf(topologyReplicaset.Alias),
+	)
+
+	// vshard group, weight, all rw
+	additionalInfo := []string{}
+	if topologyReplicaset.VshardGroup != nil {
+		additionalInfo = append(additionalInfo, *topologyReplicaset.VshardGroup)
+	}
+	if topologyReplicaset.Weight != nil {
+		formattedWeight := strconv.FormatFloat(*topologyReplicaset.Weight, 'f', -1, 64)
+		additionalInfo = append(additionalInfo, formattedWeight)
+	}
+	if topologyReplicaset.AllRW != nil && *(topologyReplicaset.AllRW) {
+		additionalInfo = append(additionalInfo, "ALL RW")
+	}
+
+	if len(additionalInfo) > 0 {
+		replicasetTitle = fmt.Sprintf(
+			"%-30s    %s",
+			replicasetTitle,
+			common.ColorHiBlue.Sprint(strings.Join(additionalInfo, " | ")),
+		)
+	}
+
+	// roles
```

Comment:
It would be great to extend comments with an example how it looks like:
```
 Role: my-role1 | my-role2
```

Of course this comment is relevant for ALL comments and places where you format servers/replicasets.

Reply:
Added comments and unit tests.